core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -970,11 +970,25 @@ private static String getCategoryDisplayName(String category) {
 
     public List<Plugin> getUpdates() {
         Map<String,Plugin> pluginMap = new LinkedHashMap<>();
+        Map<String, List<Plugin>> incompatiblePluginMap = new LinkedHashMap<>();
```

Comment:
If we would index this map with `Plugin` instead of plugin name, we could avoid lookup in `pluginMap`.

Reply:
Could you please explain this further to me?
If `incompatiblePluginMap` would be indexed with `Plugin`, how would I iterate through the `pluginMap` later and add the incompatible plugins there?